File: src/LRSignature/encode-test.php
#!/usr/local/opt/php54/bin/php
<?php

require "/Users/joehobson/Documents/navNorth/OET/LearningReg/src/LRphpLib/LearningRegistry/src/Bencode/LearningRegistryBencodeEncoder.php";

use LearningRegistryBencodeEncoder;

$startJSON = 'abcdefghijklmnopqrstuvwxyz0123456789';

$hashedText = hash('SHA256', $startJSON);
$hashedText = hash('SHA256', $startJSON);
$bencoder = new LearningRegistryBencodeEncoder($hashedText);
$bencodedText = $bencoder->encodeData($hashedText);

#hashedText = hashlib.sha256(startJSON.encode("utf-8")).hexdigest()
#hashedUTF8 = hashlib.sha256(startJSON).hexdigest()
#encoded = bencode(hashedUTF8)

# Print it
fwrite(STDOUT, "Straight is :" . $hashedText);
fwrite(STDOUT, "UTF8 is :" . $utf8_encode($hashedText));
fwrite(STDOUT, "Encoded is :" . $bencodedText);

?>

File: src/LRSignature/encode-test.py
#!/usr/local/bin/python
# -*- coding: utf-8 -*- 
import re
import hashlib
import gnupg
import types
import os, copy
from bencode import bencode

startJSON = '{"activity":{"actor":{"displayName": "Joe Hobson","id": "http://linkedin.com/in/joehobson","objectType": "Person"},"object": {"displayName": "Brokers of Expertise Standards Matchers","id": "http://myboe.org/go/groups/standards_matchers","objectType": "group"  },"verb": {"action": "endorse","content": "The Brokers of Expertise Standards Matchers group found at http://myboe.org/go/groups/standards_matchers was endorsed by Joe Hobson.","date": "2015-08-11"}}}';

encoded = bencode(startJSON)
hashedText = hashlib.sha256(encoded).hexdigest()

# Print it
print ("Encoded : %s" % encoded)
print ("hashed : %s" % hashedText)